YoCode/DuplicationCheck.cs

```diff
@@ -29,25 +29,18 @@ internal class DuplicationCheck
 
         private double passPerc = 0.5;
 
-        public DuplicationCheck(IPathManager dir, IDupFinder dupFinder, IRunParameterChecker p)
+        public DuplicationCheck(IPathManager dir, IDupFinder dupFinder, IRunParameterChecker p, string fileNameChecked)
         {
-            OrigCodeBaseCost = Int32.Parse(p.CodeBaseCost);
-            OrigDuplicateCost = Int32.Parse(p.DuplicationCost);
-
-            DuplicationEvidence.FeatureTitle = "Code quality improvement";
-            DuplicationEvidence.Feature = Feature.DuplicationCheck;
-
 
             this.dir = dir;
             this.dupFinder = dupFinder;
+            this.fileNameChecked = fileNameChecked;
 
             modifiedSolutionPath = Path.Combine(dir.ModifiedTestDirPath, fileNameChecked);
 
             try
             {
-                ExecuteTheCheck();
```

Comment:
also not sure what I need to put in here, since I don't want the DuplicationCheck to run automatically anymore.

Reply:
Move the whole try/catch to `PerformDuplicationCheck`. It isn't needed in the constructor any more but it probably is needed in the place that code moved to, for the same reasons we had it here originally.